homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -179,11 +179,15 @@ async def async_step_ssdp(self, discovery_info):
             discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME].split("(", 1)[0].strip()
         )
 
+        mac = discovery_info[ssdp.ATTR_UPNP_SERIAL].upper()
         # Synology NAS can broadcast on multiple IP addresses, since they can be connected to multiple ethernets.
         # The serial of the NAS is actually its MAC address.
-        if self._mac_already_configured(discovery_info[ssdp.ATTR_UPNP_SERIAL].upper()):
+        if self._mac_already_configured(mac):
             return self.async_abort(reason="already_configured")
 
+        await self.async_set_unique_id(mac)
```

Comment:
Isn't the serial number used for the unique id?

Reply:
Never-mind, its always going to go though `async_step_user` and be swapped out for the serial number before the entry is saved. 👍